lib/1.4/utility.dml

```diff
@@ -239,13 +345,12 @@ template _init_val_hard_reset is (init_val, hard_reset) {
 
 // Instantiate this on top level to support hard reset.
 template hreset is hard_reset {
-    port HRESET {
+    port HRESET is signal_port {
```

Comment:
This is an incompatible change, it makes HRESET less forgiving which probably will give massive failures in SSM code.

I suggest that you break out the reset parts to a separate PR and merge the new templates only, and make sure to run VP pretests before merging the reset template changes.

Reply:
Agreed, but our users are also extra sensitive to incompatibilities these days so better to play safe, and no need to delay the more urgent stuff waiting for pretests.

Reply:
Updating the reset-templates is the urgent part of this PR though, right now they're often unusable since the raise/lower methods of the signal interface can't be overridden.

Reply:
I suppose adding that `default` KW is orthogonal from adding the template, though. Switching behaviour to signal_port breaks compatibility just as much regardless.

Reply:
Yes it is, I'm just reminiscing about where all this started ;-)